Library/Homebrew/cmd/update.sh

```diff
@@ -448,7 +426,38 @@ EOS
   for DIR in "$HOMEBREW_REPOSITORY" "$HOMEBREW_LIBRARY"/Taps/*/*
   do
     [[ -d "$DIR/.git" ]] || continue
-    pull "$DIR"
+
+    # Used for testing purposes, e.g., for testing formula migration after
+    # renaming it in the currently checked-out branch. To test run
+    # "brew update --simulate-from-current-branch"
+    if [[ -n "$HOMEBREW_SIMULATE_FROM_CURRENT_BRANCH" ]]
+    then
+      UPSTREAM_BRANCH="$(upstream_branch)"
+      INITIAL_REVISION="$(git rev-parse -q --verify "$UPSTREAM_BRANCH")"
+      CURRENT_REVISION="$(read_current_revision)"
+      export HOMEBREW_UPDATE_BEFORE"$TAP_VAR"="$INITIAL_REVISION"
+      export HOMEBREW_UPDATE_AFTER"$TAP_VAR"="$CURRENT_REVISION"
```

Comment:
`$TAP_VAR` is being used here before it is populated (further below). However, I wonder if we should just remove the `--simulate-from-current-branch` option and the associated code from the update. It complicates things and makes the code a bit harder to understand and is probably rarely used, if ever.

If we wanted to retain the functionality (use local changes to test the migration code that kicks in after an update), I think this could nowadays better live in a `dev-cmd` that is custom-tailored to this use case. It could refrain from fetching and doing other modifications to the repositories and instead just set up the necessary environment variables and then invoke `brew update-report` (which will do most of the work).